src/Versionable.php

```diff
@@ -256,4 +256,40 @@ public static function withoutVersion(callable $callback)
 
         self::$versioning = true;
     }
+    /**
+     * transform the contents by versionTrans
+     *
+     * @return array
+     */
+    public function versionsTrans()
+    {
+        $data = $this->versions;
+        $data->transform(function ($item, $key) {
+            $item['contents'] = $this->transformContent($item['contents']);
+            return $item;
+        });
+        return $data;
+    }
+    /**
+     *
+     * @param array $data
+     *
+     * @return array
+     */
+    public function transformContent($data)
```

Comment:
You need check the argument type before use. And I suggest this method change to  protected function.
Replace the name meaningless variable such as $res under the function.
Please input a empty line before your return.
The principle of early exit avoids the multiple inclusion logic of large segments.